File: 1015.reversiblePrimes.cpp
/*************************************************************************
    > File Name: 1015.reversiblePrimes.cpp
    > Author: Meng jl
    > Mail:mjl8789@gmail.com 
    > Created Time: Sun 23 Feb 2014 08:59:12 PM CST
 ************************************************************************/

#include<iostream>
#include<cmath>
#include<cstdio>
#include<stack>
using namespace std;
int isprime(int n)
{
    for(int d=2;(d*d)<(n+1);++d)
    {
        if(!(n%d))
            return false;
    }
    if(n==1) return false;
    else
    return true;
}
int trans(int n,int base)
{
    stack<int> result;
    while(n)
    {
        result.push(n%base);
        n/=base;
    }
    int sum = 0;
    int pos = 0;
    while(!result.empty())
    {
        sum += result.top() * pow(base,pos++);
        result.pop();
    }
    return sum;
}
int main()
{
    int n,m;
    while(scanf("%d",&n) )
    {
        if(n<0) return 0;
        scanf("%d",&m);
        //printf("%d %d\n",n,trans(n,m));
        if(isprime(n) && isprime(trans(n,m)))
            printf("Yes\n");
        else
            printf("No\n");
    }
    return 0;
}

File: 1015.reversiblePrimes.py
import math
def isprime(a):
    if a ==2:
        return False
    if a==3:
        return True
    for i in range(2,a):
        if a%i ==0:
            return False
    return True
def trans(a,radix):
    total =[]
    sum =0
    while a:
        total.append(a%radix)
        a = a /radix
    for i in range(len(total)):
        sum = sum + total[i]*pow(radix,i)


lists = [int(x) for x in raw_input().split()]
while len(lists) == 2:
    if isprime(lists[0]) and isprime(trans(lists[0],lists[1])):
        print "Yes"
    else:
        print "No"
    lists = [int(x) for x in raw_input().split()]
